src/rules/no-restricted-paths.js

```diff
@@ -56,78 +56,125 @@ module.exports = {
-      let reportInvalidException;
 
-      if (isGlobPattern) {
-        const mm = new Minimatch(absoluteFrom);
-        isPathRestricted = (absoluteImportPath) => mm.match(absoluteImportPath);
+      const mm = new Minimatch(absoluteFrom);
+      const isPathRestricted = (absoluteImportPath) => mm.match(absoluteImportPath);
+      const hasValidExceptions = zoneExcept.every(isGlob);
 
-        hasValidExceptions = zoneExcept.every(isGlob);
+      if (hasValidExceptions) {
+        const exceptionsMm = zoneExcept.map((except) => new Minimatch(except));
+        isPathException = (absoluteImportPath) => exceptionsMm.some((mm) => mm.match(absoluteImportPath));
+      }
 
-        if (hasValidExceptions) {
-          const exceptionsMm = zoneExcept.map((except) => new Minimatch(except));
-          isPathException = (absoluteImportPath) => exceptionsMm.some((mm) => mm.match(absoluteImportPath));
-        }
+      const reportInvalidException = reportInvalidExceptionGlob;
 
-        reportInvalidException = reportInvalidExceptionGlob;
-      } else {
-        isPathRestricted = (absoluteImportPath) => containsPath(absoluteImportPath, absoluteFrom);
+      return {
+        isPathRestricted,
+        hasValidExceptions,
+        isPathException,
+        reportInvalidException,
+      };
+    }
 
-        const absoluteExceptionPaths = zoneExcept
-          .map((exceptionPath) => path.resolve(absoluteFrom, exceptionPath));
-        hasValidExceptions = absoluteExceptionPaths
-          .every((absoluteExceptionPath) => isValidExceptionPath(absoluteFrom, absoluteExceptionPath));
+    function computeAbsolutePathValidator(absoluteFrom, zoneExcept) {
+      let isPathException;
 
-        if (hasValidExceptions) {
-          isPathException = (absoluteImportPath) => absoluteExceptionPaths.some(
-            (absoluteExceptionPath) => containsPath(absoluteImportPath, absoluteExceptionPath),
-          );
-        }
+      const isPathRestricted = (absoluteImportPath) => containsPath(absoluteImportPath, absoluteFrom);
 
-        reportInvalidException = reportInvalidExceptionPath;
+      const absoluteExceptionPaths = zoneExcept
+        .map((exceptionPath) => path.resolve(absoluteFrom, exceptionPath));
+      const hasValidExceptions = absoluteExceptionPaths
+        .every((absoluteExceptionPath) => isValidExceptionPath(absoluteFrom, absoluteExceptionPath));
+
+      if (hasValidExceptions) {
+        isPathException = (absoluteImportPath) => absoluteExceptionPaths.some(
+          (absoluteExceptionPath) => containsPath(absoluteImportPath, absoluteExceptionPath),
+        );
       }
 
+      const reportInvalidException = reportInvalidExceptionPath;
+
       return {
         isPathRestricted,
         hasValidExceptions,
         isPathException,
         reportInvalidException,
       };
+    }
+
+    const makePathValidators = (zoneFrom, zoneExcept = []) => {
+      const allZoneFrom = zoneFrom instanceof Array ? zoneFrom : [zoneFrom];
```

Comment:
```suggestion
      const allZoneFrom = [].concat(zoneFrom);
```

Reply:
@ljharb good point 👍🏽 
fixed in https://github.com/import-js/eslint-plugin-import/pull/2466/commits/95d16861d1ec2c0f0902b57330356857ee152540